lib/luahid.c

```diff
@@ -93,6 +93,83 @@ static const struct hid_device_id *luahid_setidtable(lua_State *L, int idx)
 	return user_table;
 }
 
+#define luahid_setfield(L, idx, obj, field)	\
+do { 						\
+	lua_pushinteger(L, obj->field);		\
+	lua_setfield(L, idx, #field);		\
+} while (0)
+
+static inline void luahid_pushdevid(lua_State *L, int idx, const struct hid_device_id *device_id)
+{
+	lua_newtable(L);
+	luahid_setfield(L, idx, device_id, bus);
+	luahid_setfield(L, idx, device_id, group);
+	luahid_setfield(L, idx, device_id, vendor);
+	luahid_setfield(L, idx, device_id, product);
+	luahid_setfield(L, idx, device_id, driver_data);
+}
+
+static int luahid_probe_handler(lua_State *L, luahid_t *hid, struct hid_device *hdev, const struct hid_device_id *id)
+{
+	int ret = 0;
+
+	if (lunatik_getregistry(L, hid) != LUA_TTABLE ||
+	    lua_getfield(L, -1, "driver") != LUA_TTABLE) {
+		pr_err("probe: couldn't find driver");
+		ret = -ENXIO;
+		goto out;
+	}
+
+	if (lua_getfield(L, -1, "probe") != LUA_TFUNCTION) {
+		ret = 0;
+		goto out;
+	}
+
+	lua_pushvalue(L, -2); /* driver table */
+	luahid_pushdevid(L, -2, id);
+
+	if (lua_pcall(L, 2, 1, 0) != LUA_OK) {
+		pr_err("Failed to call init function: %s", lua_tostring(L, -1));
```

Comment:
please, review all your log messages and try to base them on luadevice.. e.g., https://github.com/luainkernel/lunatik/blob/master/lib/luadevice.c#L109
```suggestion
		pr_err("probe: %s\n", lua_tostring(L, -1));
```

also, please, avoid upper case on the beginning of the log message

Reply:
changed